CriticalMass/LocationProvider.swift

```diff
@@ -17,4 +17,8 @@ public enum LocationProviderPermission: String {
 public protocol LocationProvider {
     var currentLocation: Location? { get }
     static var accessPermission: LocationProviderPermission { get }
+
+    func updateLocation()
+    var locationUpdateHandler: (() -> Void)? { get set }
+    var locationErrorHandler: (() -> Void)? { get set }
```

Comment:
Minor: maybe we can pass a completion block into updateLocation instead of exposing the blocks publicly. We also could use Result here similar to ResultCallback

Reply:
Nice idea, I changed updateLocation() to method with completion block